### PR TITLE
Data Flow: Deprecate old data flow api.

### PR DESCRIPTION
**Pull request by aschackmull**

### CODE REVIEW

**Comment by jketema** [comment]
Fix for the coding standards tests: https://github.com/github/codeql-coding-standards/pull/468

File: java/ql/lib/semmle/code/java/dataflow/internal/DataFlowImpl1.qll [hunk 1]
```diff
@@ -14,6 +14,8 @@ import FlowStateString
 private import codeql.util.Unit
```
[the remark this answers is not in the document]
Comment (aschackmull) [reply java/ql/lib/semmle/code/java/dataflow/internal/DataFlowImpl1.qll:14]: Agree, but I figured that was slightly more likely to break something, so I initially left it out, in order to hopefully merge this faster. Then deprecation of `FlowStateString` can follow in another PR.

**Review by jketema — APPROVED** [review]

**Review by hvitved — APPROVED** [review]

**Review by jketema — APPROVED** [review]

**Review by michaelnebel — APPROVED** [review]
LGTM!

**Comment by jketema** [comment]
This probably still needs some change notes?

**Pull request merged**